lib/route/src/Obelisk/Route/Frontend.hs

```diff
@@ -530,14 +550,17 @@ routeLinkImpl
      , RouteToUrl route m
      , SetRoute t route m
      )
-  => route -- ^ Target route
+  => Map AttributeName Text
+  -> route -- ^ Target route
   -> m a -- ^ Child widget
   -> m (Event t (), a)
-routeLinkImpl r w = do
+routeLinkImpl attrs r w = do
   enc <- askRouteToUrl
   let cfg = (def :: ElementConfig EventResult t (DomBuilderSpace m))
-        & elementConfig_eventSpec %~ addEventSpecFlags (Proxy :: Proxy (DomBuilderSpace m)) Click (\_ -> preventDefault)
-        & elementConfig_initialAttributes .~ "href" =: enc r
+          & elementConfig_initialAttributes .~ ("href" =: enc r <> attrs)
+          & (if Map.lookup "target" attrs == Just "_blank"
+             then id
+             else elementConfig_eventSpec %~ addEventSpecFlags (Proxy :: Proxy (DomBuilderSpace m)) Click (\_ -> preventDefault))
```

Comment:
In the case of _blank we actually want to not even `setRoute` below either.

Reply:
Actually we *also* don't want to `scrollToTop`

Reply:
Thanks, done